openedx/core/djangoapps/content_tagging/api.py

```diff
@@ -160,19 +145,32 @@ def tag_content_object(
 
     Raises ValueError if the proposed tags are invalid for this taxonomy.
     Preserves existing (valid) tags, adds new (valid) tags, and removes omitted (or invalid) tags.
+
+    We require that this taxonomy is linked to the content object's "org" or linked to "all orgs" (None).
     """
+    object_key: UsageKey | CourseKey
+
     if not taxonomy.system_defined:
         # We require that this taxonomy is linked to the content object's "org" or linked to "all orgs" (None):
+        try:
+            object_key = UsageKey.from_string(object_id)
+        except InvalidKeyError:
+            try:
+                object_key = CourseKey.from_string(object_id)
+            except InvalidKeyError as e:
+                raise ValueError("object_id must be from a block or a course") from e
```

Comment:
Why is this validation nested under `if not taxonomy.system_defined:`?
I would think we should always validate the object_key and org, even for system taxonomies.

Reply:
Fixed!